sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -2108,7 +2109,7 @@ def Batch(
             )
         if response_hook:
             response_hook(last_response_headers, final_responses)
-        return final_responses
+        return CosmosListResponse(original_list=final_responses, response_headers=last_response_headers)
```

Comment:
Depending on how we want to document this, we may want to update the return types for all the methods appropriately (e.g. `dict[str, Any]` -> `CosmosDict[str, Any]`
As otherwise customers may not necessarily find this new attribute. But I don't think we need to for a preview release.
We may also need to make these objects public if we want them to appear as part of the rendered documentation.

Reply:
if we change the type on a public doc it will be counted as a breaking change right?